content/blog/2022/01/mutating-policy-behave-as-validating.md

```diff
@@ -0,0 +1,41 @@
+---
+title: Kubewarden has policies to cover all the Kubernetes Pod Security Policies.
+authors:
+	- José Guilherme Vanz
+date: 2022-01-31
+---
+
+We at the Kubewarden team has been doing an effort to allow our users to have
+policies equivalent for all the deprecated Pod Security Policies (PSP).
+So, we wrote policies to replicate the same validations available in the Kubernetes
+PSPs. During this work, we counted on the community help to map and validate
+the policies.
+
+Since Kubewarden controller <version-placeholder> and Policy Server version
+<version-placeholder>, Kubewarden users have policies which replace all the PSPs!
+All the them can be found in the Policy Hub looking for the keyword "PSP".
+
+You may are thinking: why do I need to a specific versions to have all PSP policies?
+
+Actually, depending on what you would like to do, you do not need that. Let me
+explain.
+
+Before the version mentioned previously in this text. If a policy mutate a
+request and its "mutating" configuration field is 'true', the policy server
+always accept the request.  However, this is not in-line with PSP. Users
+should be able to reject a request if a mutation happen. To solve this issue,
+we change the Kubewarden controller and Policy Server to allow the user
+configure a mutating policy to behave like a validating one. If you do not need
+that, you can use the policies in previous Kubewarden versions.
+
+Since these changes, when the user mark a policy as "mutating" (setting 'true'
+in the "mutating" field in the ClusterAdmissionPolicy resource) everything work
+as the before, mutated requests will be accepted. But when the user deploy a
+policy which mutates requests and set the "mutating" field to `false`, mutated
+requests are going to be rejected. This means that Kubewarden now not only have
+policies which cover 100% of the deprecated PSPs, but they behave the same of
+what a PSP users expect.
```

Comment:
```suggestion
Since the Kubewarden controller and Policy Server versions update, if you mark a policy as "mutating" (setting 'true'
in the "mutating" field in the ClusterAdmissionPolicy resource), the mutated requests will be accepted. And if you deploy a policy which mutates requests with the "mutating" field set to `false`, the requests will be rejected. 

This means Kubewarden now have policies which cover 100% of the deprecated PSPs, and they behave the same way than the PSP policies.
```